src/mds/mdstypes.h

```diff
@@ -1545,6 +1545,7 @@ class MDSCacheObject {
   // waiting
  protected:
   compact_multimap<uint64_t, MDSInternalContextBase*>  waiting;
+  static const uint64_t WAIT_LOCK_MASK = ~0xffull;
```

Comment:
This constant needs a comment explaining where it comes from.

Maybe it should also be defined somewhere in the same place as the flags it's detecting to make it clearer?